tests/phpunit/tests/html-api/wpHtmlTagProcessor.php

```diff
@@ -2939,4 +2939,45 @@ public function test_unclosed_funky_comment_input_too_short() {
 		$this->assertFalse( $processor->next_tag() );
 		$this->assertTrue( $processor->paused_at_incomplete_token() );
 	}
+
+	/**
+	 * Test doctype name handling.
+	 *
+	 * @dataProvider data_doctypes
+	 *
+	 * @ticket 61576
+	 */
+	public function test_doctype_doc_name( string $html, ?string $expected_doctype_name ) {
```

Comment:
There are good doctype tests as part of the HTML5lib suite and they caught a few issues I had.

We can consider adding more tests here or some specific tests for the behavior of the doctype_info class, especially regarding get_compatibility_mode although even that is covered by the HTML5lib tests where they often use `<p><table>` as a test for whether quirks mode is set correctly.